src/supercollider-js/SuperColliderJS.sc

```diff
@@ -20,12 +41,15 @@ SuperColliderJS {
 
 		thisProcess.nowExecutingPath = executingPath;
 
-		// capture compile errors, stdout
-		"\nSUPERCOLLIDERJS:%:CAPTURE:START\n".format(guid).postln;
+		// capture compile errors - they're written directly to STDOUT and don't
+		// throw a catchable exception. Also capture any STDOUT written by the
+		// code we're calling. We'll also capture anything else that happens to
+		// get written to STDOUT while we're capturing, but c'est la vie
+		"SUPERCOLLIDERJS:%:CAPTURE:START".format(guid).postln;
 		compiled = code.compile;
 
 		if(compiled.isNil, {
-			"\nSUPERCOLLIDERJS:%:CAPTURE:END".format(guid).postln;
+			"SUPERCOLLIDERJS:%:CAPTURE:END".format(guid).postln;
```

Comment:
The newlines are actually necessary in case sclang posted, not postln.  The SUPERCOLLIDERJS messages need to always start on a newline.